src/main/dialogs/basic/dialog.js

```diff
@@ -183,13 +194,48 @@ fluid.defaults("gpii.app.dialog", {
             funcName: "gpii.app.dialog.registerDailogReadyListener",
             args: "{that}"
         },
+        "onCreate.applyScaleFactor": {
+            funcName: "gpii.app.dialog.rescaleDialog",
+            args: ["{that}", "{that}.model.scaleFactor"]
+        },
         "onDestroy.cleanupElectron": {
             this: "{that}.dialog",
             method: "destroy",
             priority: "last"
         }
     },
     invokers: {
+        getScaledWidth: {
```

Comment:
Rather than 3 invokers, might be neater to just put the scaled variants of these values into the model alongside the unscaled ones using a relay, then they can just be read like other values.

Reply:
This comment still outstanding

Reply:
In the current model we already keep the scaled variants of the metrics (that’s the reason for using the old `scaleFactor` when calculating the rescaled metrics). These three methods simply recompute the metrics on a change in the scaling factor… so probably a name such as `getRescaled<Metric>` would suit them better.
 
The simplest solutions I can think of that would fulfill your comment are:
* In order to have the scaled versions together with the unscaled ones we might simply compute the unscaled versions based on the updates of the currently existing scaled properties (width, height and offset). It would further follow your suggestion. However, I would prefer a different approach.
* On the other hand, without the unscaled versions we might simply have a `modelRelay` that watches for changes in the `scaleFactor` and updates the metrics accordingly. In combination with a `modelListener` for the `scaleFactor` (assuming the `modelRelay` executes first) that applies the newly computed metrics and updates the zooming of the corresponding BrowserWindow this should do it.  

How do these two sound to you, am I following your idea correctly?
 
Alternatively, the overall scaling mechanism could be re-engineered so that we make use of both the scaled and unscaled versions of the metrics.

Reply:
OK, now I understand properly that this is an instantaneous, historyful interaction I think we should not try to involve the model system in this. I suggest instead we just consolidate these three invokers into a single one which accepts a {width:, height:, offset} structure (thus avoiding wholesale dependence on "that") and returns a scaled version of it, given these appear to always be requested simultaneously.